src/validation.cpp

```diff
@@ -1528,14 +1528,20 @@ bool CheckInputScripts(const CTransaction& tx, TxValidationState &state, const C
         return true;
     }
 
-    if (!txdata.m_ready) {
-        txdata.Init(tx);
+    if (!txdata.m_spent_outputs_ready) {
+        std::vector<CTxOut> spent_outputs;
+        spent_outputs.reserve(tx.vin.size());
+
+        for (unsigned int i = 0; i < tx.vin.size(); ++i) {
+            const COutPoint &prevout = tx.vin[i].prevout;
```

Comment:
Any reason not to do a range-based for loop here?

```suggestion
        for (const CTxIn& txin : tx.vin) {
            const COutPoint &prevout = txin.prevout;
```

Reply:
Done.